examples/structural_mechanics/crash/data_transformations.py

```diff
@@ -0,0 +1,151 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import numpy as np
+
+from physicsnemo_curator.etl.data_transformations import DataTransformation
+from physicsnemo_curator.etl.processing_config import ProcessingConfig
+
+from .schemas import CrashExtractedDataInMemory
+
+
+class CrashDataTransformation(DataTransformation):
+    """Transform crash simulation data: filter nodes, compute thickness, build edges."""
+
+    def __init__(
+        self,
+        cfg: ProcessingConfig,
+        wall_threshold: float = 1.0,
+    ):
+        super().__init__(cfg)
+        self.wall_threshold = wall_threshold
+
+    def transform(self, data: CrashExtractedDataInMemory) -> CrashExtractedDataInMemory:
```

Comment:
This looks good with the known concern that this code is a refactored code from the example (e.g. from `process_d3plot_data` and other functions). So we have 2 different implementations which do the same, so merging the changes can be a challenge.
But I don't see a good alternative.